pulp_rpm/app/tasks/synchronizing.py

```diff
@@ -174,19 +174,15 @@ def add_metadata_to_publication(publication, version, prefix=""):
     published_artifacts = []
 
     # Handle packages
-    pkg_data = (
-        ContentArtifact.objects.filter(
-            content__in=version.content, content__pulp_type=Package.get_pulp_type()
-        )
-        .select_related("content__rpm_package")
-        .only("pk", "artifact", "content", "content__rpm_package__pkgId")
-    )
+    pkg_data = ContentArtifact.objects.filter(
+        content__in=version.content, content__pulp_type=Package.get_pulp_type()
+    ).values("pk", "content__rpm_package__pkgId")
     for ca in pkg_data.iterator():
         for relative_path in pkgid_to_location_href[str(version.repository.pk)][
-            ca.content.rpm_package.pkgId
+            ca["content__rpm_package__pkgId"]
         ]:
             pa = PublishedArtifact(
-                content_artifact=ca,
+                content_artifact_id=ca["pk"],
```

Comment:
I wasn't aware that this worked, nice

Reply:
Since django stores foreign keys id in the field formed by field_name + _id, we can pass the foreign key id directly to that field without having to pass the whole object given that we already have the id of it.